src/app/fdctl/run/tiles/fd_shred.c

```diff
@@ -540,74 +600,84 @@ populate_packet_header_template( eth_ip_udp_t * pkt,
   pkt->udp->check     = (ushort)0;
 }
 
-#define NONNULL( x ) (__extension__({                                       \
-      __typeof__((x)) __x = (x);                                            \
-      if( FD_UNLIKELY( !__x ) ) FD_LOG_ERR(( #x " was unexpectedly NULL" )); \
-      __x; }))
+static void
+privileged_init( fd_topo_t *      topo,
+                 fd_topo_tile_t * tile,
+                 void *           scratch ) {
+  (void)topo;
 
-int
-fd_shred_tile( fd_shred_tile_args_t * args, void * scratch ) {
-  shred_tile_ctx_t ctx[1];
-
-  ulong bank_cnt = args->bank_cnt;
-
-  NONNULL( args->cnc);
-  NONNULL( args->rng );
-  NONNULL( args->ip  );
-  NONNULL( args->cluster_nodes_mvcc  );
-  for( ulong i=0UL; i<bank_cnt; i++ ) {
-    NONNULL( args->bank_shred_mcache[ i ] );
-    NONNULL( args->bank_shred_fseq  [ i ] );
-    NONNULL( args->bank_shred_chunk0[ i ] );
-    NONNULL( args->bank_shred_wmark [ i ] );
-  }
-  NONNULL( args->bank_shred_wksp );
-
-  NONNULL( args->from_net_mcache );
-  NONNULL( args->from_net_fseq   );
-  NONNULL( args->from_net_chunk0 );
-  NONNULL( args->from_net_wmark  );
-
-  NONNULL( args->to_net_mcache );
-  NONNULL( args->to_net_dcache );
-  NONNULL( args->to_net_fseq   );
-
-  NONNULL( args->shred_store_mcache );
-  NONNULL( args->shred_store_dcache );
-  NONNULL( args->shred_store_fseq   );
-
-  fd_mux_callbacks_t callbacks[1] = { 0 };
-  callbacks->during_housekeeping = during_housekeeping;
-  callbacks->before_frag         = before_frag;
-  callbacks->during_frag         = during_frag;
-  callbacks->after_frag          = after_frag;
-
-  ulong shred_store_mcache_depth = fd_mcache_depth( args->shred_store_mcache );
-  ulong fec_resolver_footprint = fd_fec_resolver_footprint( args->fec_resolver_depth, 1UL, shred_store_mcache_depth,
-                                                            args->fec_resolver_done_depth );
+  ulong scratch_top = (ulong)scratch;
+  fd_shred_init_ctx_t * init_ctx = (fd_shred_init_ctx_t*)SCRATCH_ALLOC( alignof( fd_shred_init_ctx_t ), sizeof( fd_shred_init_ctx_t ) );
```

Comment:
If we can write to scratch from here, can we not just write those fields of the `fd_shred_ctx_t`?